src/storage/v3/shard_rsm.cpp

```diff
@@ -112,6 +124,248 @@ Value ConstructValueVertex(const memgraph::storage::v3::VertexAccessor &acc, mem
+  ctx.properties = NamesToProperties(storage.properties_, dba);
+  ctx.labels = NamesToLabels(storage.labels_, dba);
+  auto value = expr->Accept(eval);
+  return value;
+}
+
+std::vector<PropertyId> GetPropertiesFromAcessor(
+    const std::map<memgraph::storage::v3::PropertyId, memgraph::storage::v3::PropertyValue> &properties) {
+  std::vector<PropertyId> ret_properties;
+  ret_properties.reserve(properties.size());
+
+  for (const auto &prop : properties) {
+    ret_properties.emplace_back(prop.first);
+  }
+
+  return ret_properties;
+}
+
+memgraph::expr::EvaluationContext CreateEvaluationContext(
+    const std::optional<memgraph::storage::v3::VertexAccessor> &v_acc,
+    const std::optional<memgraph::storage::v3::EdgeAccessor> &e_acc) {
+  memgraph::expr::EvaluationContext ctx;
+
+  std::vector<PropertyId> vertex_props;
+  std::vector<PropertyId> edge_props;
+  std::vector<memgraph::storage::v3::LabelId> labels_in_shard;
+
+  if (v_acc) {
+    const auto labels = v_acc->Labels(View::OLD);
+    labels_in_shard.reserve(labels.GetValue().size());
+
+    for (const auto &label : labels.GetValue()) {
+      labels_in_shard.emplace_back(label);
+    }
+
+    vertex_props = GetPropertiesFromAcessor(v_acc->Properties(View::OLD).GetValue());
+  }
+  if (e_acc) {
+    edge_props = GetPropertiesFromAcessor(e_acc->Properties(View::OLD).GetValue());
+  }
+
+  std::vector<PropertyId> all_properties(vertex_props.size() + edge_props.size());
+
+  // Sorting might not needed, since the props are being converted from std::map
+  std::sort(vertex_props.begin(), vertex_props.end());
+  std::sort(edge_props.begin(), edge_props.end());
+
+  auto it = std::set_symmetric_difference(vertex_props.begin(), vertex_props.end(), edge_props.begin(),
+                                          edge_props.end(), all_properties.begin());
+  all_properties.resize(it - all_properties.begin());
+
+  ctx.properties = std::move(all_properties);
+  ctx.labels = std::move(labels_in_shard);
+
+  return ctx;
+}
+
+std::any ParseExpression(const std::string &expr, memgraph::expr::AstStorage &storage) {
+  memgraph::frontend::opencypher::Parser<memgraph::frontend::opencypher::ParserOpTag::EXPRESSION> parser(expr);
+  memgraph::expr::ParsingContext pc;
+  memgraph::expr::CypherMainVisitor visitor(pc, &storage);
+
+  auto *ast = parser.tree();
+  return visitor.visit(ast);
+}
+
+memgraph::expr::TypedValue ComputeExpression(memgraph::expr::DbAccessor &dba,
+                                             const std::optional<memgraph::storage::v3::VertexAccessor> &v_acc,
+                                             const std::optional<memgraph::storage::v3::EdgeAccessor> &e_acc,
+                                             const std::string &expression, std::string_view node_name,
+                                             std::string_view edge_name) {
+  memgraph::expr::AstStorage storage;
+  memgraph::expr::Frame<memgraph::expr::TypedValue> frame{static_cast<int64_t>(128)};
```

Comment:
128 here is extremely suspicious, shouldn't this be determined based on the actual expression?

Reply:
@gitbuda 
Yes, the `128` is a leftover I forgot to remove.
We know 2 things at this stage, at no point, should the size of the `Frame` in this context be bigger than 2. At most it will be one `Vertex` and one `Edge` on the Frame. The way the signature is written, both the `VertexAccessor` and the `EdgeAccessor` is optional, based on that we can determine if the size of the frame should be 0, 1 or 2. I'm not sure if 0 is a common use-case though.
Based on that I can set the size of the frame, or just hard-code it to 2. (I'm not sure if the memory consumption would be significant).
What do you think?